### PR TITLE
fix(api): prevent saving variants without any conditions

### PR DESCRIPTION
**Pull request by antonjoel82**
### What change does this PR introduce?

- Server-side validation of variant filters / conditions via a new util function

### Why was this change needed?

- Follow-up to [NV-3147](https://linear.app/novu/issue/NV-3147/%F0%9F%94%80-able-to-save-variants-without-any-conditions)

### Other information (Screenshots)

_The following screengrab was recorded by temporarily removing front-end validations (by deleting `shouldDisallowEmptyConditions` in `VariantItemCard.tsx`)_

https://www.loom.com/share/f3497c33507541ffad4aae82e473c38b?sid=dd54460a-5acd-42eb-a6d0-1fca87eb3c07

### CODE REVIEW

**Review by ainouzgali — APPROVED** [review]
Looks good🚀

File: apps/api/src/app/workflows/utils/checkIsVariantEmpty.spec.ts [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    const emptyVariant: NotificationStepVariant = { filters: [] };
+    const result = checkIsVariantEmpty(emptyVariant);
+    expect(result).to.be.true;
+  });
+
+  it('should return true for a variant with empty filters', () => {
+    const variantWithEmptyFilters: NotificationStepVariant = { filters: [] };
+    const result = checkIsVariantEmpty(variantWithEmptyFilters);
+    expect(result).to.be.true;
+  });
```
Comment (LetItRock) [inline apps/api/src/app/workflows/utils/checkIsVariantEmpty.spec.ts:22]: same test code ;)

File: apps/api/src/app/workflows/utils/checkIsVariantEmpty.ts [hunk 2]
```diff
@@ -0,0 +1,6 @@
+import { NotificationStepVariant } from '../usecases/create-notification-template';
+
+/** determine if the variant has no filters / conditions */
+export const checkIsVariantEmpty = (variant: NotificationStepVariant): boolean => {
+  return !(variant.filters?.length && variant.filters.some((filter) => filter.children?.length));
```
Comment (LetItRock) [inline apps/api/src/app/workflows/utils/checkIsVariantEmpty.ts:5]:
```suggestion
  return !(variant.filters?.some((filter) => filter.children?.length));
```

File: apps/api/src/app/workflows/utils/checkIsVariantEmpty.spec.ts (hunk 1 above)
Comment (antonjoel82) [reply apps/api/src/app/workflows/utils/checkIsVariantEmpty.spec.ts:22]: Good catch, thank you!

**Pull request merged**